src/sortStudents.ts

```diff
@@ -1,16 +1,56 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grade',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+const getAverageGrade = ({ grades }: Student): number => {
+  return grades.reduce(
+    (firstMark, secondMark) => firstMark + secondMark, 0,
+  ) / grades.length;
+};
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copyStudents = [...students];
+
+  return copyStudents.sort((currentStudent, nextStudent) => {
+    switch (sortBy) {
+      case SortType.Name:
+      case SortType.Surname:
+        return order === 'asc'
+          ? currentStudent[sortBy].localeCompare(nextStudent[sortBy])
+          : nextStudent[sortBy].localeCompare(currentStudent[sortBy]);
```

Comment:
Little hint: instead of having two similar (swapped) code lines you can have one, but multiply it by -1 in case of `desc` order.
```suggestion
        return currentStudent[sortBy].localeCompare(nextStudent[sortBy]) * (order === 'asc' ? 1 : -1);
```

Of course, you can create a variable for this to use in all cases.